src/App.tsx

```
@@ -1,27 +1,46 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleShowAll = () => {
+    getAll().then(data => setGoods(data));
```

handleShowAll calls getAll() but does not catch or handle possible errors. This violates the checklist item: "3. [API] - When you make a request to the server, don't forget to handle possible errors". Add error handling (e.g., use async/await with try/catch or append .catch) and update UI state so failures are surfaced instead of failing silently.